python-stdlib/pathlib/pathlib.py

```diff
@@ -184,6 +184,13 @@ def with_suffix(self, suffix):
         index = -len(self.suffix) or None
         return Path(self._path[:index] + suffix)
 
+    def expanduser(self):
+        if self._path == "~" or self._path.startswith("~/"):
+            return Path(os.getenv("HOME") + self._path[1:])
+        if self._path[0] == "~":
```

Comment:
CPython doesn't raise an error in this case, maybe just leave this out? Or what was the intent (all code paths should probably be unittested)?